security/backends/elasticsearch/writer.py

```diff
@@ -376,36 +353,51 @@ def set_stale_celery_task_log_state(self):
             if task_last_run and task_last_run.state == CeleryTaskRunLogState.SUCCEEDED:
-                task.update(
-                    state=CeleryTaskInvocationLogState.SUCCEEDED,
-                    stop=task_last_run.stop,
-                    update_only_changed_fields=True
+                self.get_data_writer().update_index(
+                    task, state=CeleryTaskInvocationLogState.SUCCEEDED, stop=task_last_run.stop
                 )
             elif task_last_run and task_last_run.state == CeleryTaskRunLogState.FAILED:
-                task.update(
-                    state=CeleryTaskInvocationLogState.FAILED,
-                    stop=task_last_run.stop,
-                    update_only_changed_fields=True
+                self.get_data_writer().update_index(
+                    task, state=CeleryTaskInvocationLogState.FAILED, stop=task_last_run.stop
                 )
             else:
                 try:
+                    task_args = json.loads(task.task_args),
+                    task_kwargs = json.loads(task.task_kwargs),
                     current_app.tasks[task.name].expire_invocation(
                         task.id,
-                        json.loads(task.task_args),
-                        json.loads(task.task_kwargs),
+                        task_args,
+                        task_kwargs,
                         dict(
-                            start=task.start,
-                            celery_task_id=task.celery_task_id,
-                            stop=now(),
-                            name=task.name,
-                            queue_name=task.queue_name,
+                            slug=task.slug,
+                            parent_key=task.parent_log,
+                            related_objects=[
+                                get_object_triple_from_key(key) for key in task.related_objects or ()
+                            ],
+                            data=dict(
+                                start=task.start,
+                                celery_task_id=task.celery_task_id,
+                                stop=now(),
+                                name=task.name,
+                                queue_name=task.queue_name,
+                                applied_at=task.applied_at,
+                                triggered_at=task.triggered_at,
+                                is_unique=task.is_unique,
+                                is_async=task.is_async,
+                                is_duplicate=task.is_duplicate,
+                                is_on_commit=task.is_on_commit,
+                                input=task.input,
+                                task_args=task_args,
+                                task_kwargs=task_kwargs,
+                                estimated_time_of_first_arrival=task.estimated_time_of_first_arrival,
+                                expires_at=task.expires_at,
+                                stale_at=task.stale_at,
```

Comment:
wee need set logger object with all stored data, because logstash rewrite all not send data with nulls (use only insert not update)